sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -68,6 +68,106 @@ std::string getDeviceTypeName(const device &Device) {
   }
 }
 
+namespace syclex = sycl::ext::oneapi::experimental;
+
+const char *getArchName(const device &Device) {
+  try {
+    auto arch = Device.get_info<syclex::info::device::architecture>();
+    switch (arch) {
+#define __ARCH(ARCH)                                                           \
+  case syclex::architecture::ARCH:                                             \
```

Comment:
I think we should refactor this into a `.inc` file that would be used both here and in `device_info.hpp`, so that updates would need to be made in a single place.